### DISCUSSION TITLE
Header Widgets on Pages not aligned in Grid correctly

### DISCUSSION THREAD

**Discussion by simonjenny** [filamentphp/filament#10585] [Help] [answered]
I want to add twoStat  Widgets to the Header of a custom Page
```php
protected function getHeaderWidgets(): array
{
    return [
        AccountUsage::class,
        SendUsage::class
    ];
}

public function getHeaderWidgetsColumns(): int | array
{
    return 2;
}
```
But instead of aligning side by side they align stacked

![grafik](https://github.com/filamentphp/filament/assets/459369/77a96998-0789-404e-b128-1a77b0db6c9a)

I tried everything I found on the web 
- Overriding getColumns() in the Widget
- Overriding $columnSpan in the Widget

but no changes.

What am I missing?

**Comment by simonjenny** [comment, marked as answer]
Alright I think I found it. Sorry but IMHO the Filament Documentations are .. not good.

So to create this:

![grafik](https://github.com/filamentphp/filament/assets/459369/7d389751-d4e2-4138-a83a-0e39c1876c1c)

I had to do a combination of:

1. On the Page

```php
    public function getHeaderWidgetsColumns(): int
    {
        return 12;
    }
```

2. In every Widget

```php
    protected int | string | array $columnSpan = '6';
    protected function getColumns(): int
    {
        return 6;
    }
```

I am documenting this here for future Genrations ;)

**Reply by badrshs** [reply]
This worked for me, I've spent on this issues hours and I wasn't able to fix it, and 100% agree, filament has lots of lack  on their doc

**Reply by nul0op** [reply]
Thanks, i also spent too much time on this.
worth nothing that, let's say you want to align 4 widgets,  use:
- on the page => getHeaderWidgetsColumns = 4
- on the widget => 
  - getColumns = 1
  - protected int | string | array $columnSpan = '1';
